File: src/solution_java/easy/_69Sqrtx.java
package solution_java.easy;

public class _69Sqrtx {
public class Solution {
    public int mySqrt(int x) {
        if (x < 2) {
            return x;
        }
        int left = 0;
        int right = x;
        int ans = 0;
        while (left <= right) {
            int middle = ((right - left) >> 1) + left;
            if (x / middle >= middle) {
                ans = middle;
                left = middle + 1;
            } else {
                right = middle - 1;
            }
        }
        return ans;
    }
}
}

File: src/easy/_69SqrtX.kt
package easy

class _69SqrtX {
class Solution {
    fun mySqrt(x: Int): Int {
        if (x < 2) {
            return x
        }
        var left = 1
        var right = x
        var ans = 0
        while (left <= right) {
            val middle = left + (right - left) / 2
            if (x / middle >= middle) {
                ans = middle
                left = middle + 1
            } else {
                right = middle - 1
            }
        }
        return ans
    }
}
}